test/WidgetsTest.hs

```diff
@@ -0,0 +1,23 @@
+module WidgetsTest(tests) where
+
+import qualified Test.Tasty as T
+import qualified Test.Tasty.HUnit as HU
+
+import qualified Widgets as W
+
+
+tests :: T.TestTree
+tests = T.testGroup "Widgets Tests" [
+    HU.testCase "Percentile" percentileTest,
+    HU.testCase "Get int rank percentile" getIntRankPercentileTest,
+    HU.testCase "Get frac rank percentile" getFracRankPercentileTest
+    ]
+
+percentileTest :: HU.Assertion
+percentileTest = HU.assertEqual "Percentile" 2.5 (W.percentile 67 [1.0, 2.0, 3.0])
+
+getIntRankPercentileTest :: HU.Assertion
+getIntRankPercentileTest = HU.assertEqual "Get int rank percentile" 1.0 (W.getIntRankPercentile 0.9 [1.0, 2.0, 3.0])
```

Comment:
```suggestion
getIntRankPercentileTest = HU.assertEqual "Get int rank percentile" 2.0 (W.getIntRankPercentile 1.0 [1.0, 2.0, 3.0])
```
Supposed to be an integer `r` that is passed for `getIntRankPercentile`. I'll update the type signature later to make this clear